tools/clang/lib/SPIRV/TypeTranslator.cpp

```diff
@@ -155,6 +155,32 @@ bool TypeTranslator::is1xNMatrixType(QualType type) {
   return rowCount == 1 && colCount > 1;
 }
 
+bool TypeTranslator::isScalarMatOrVectorMat(QualType type) {
+  return is1x1MatrixType(type) || isSingleRowOrSingleColMatrix(type);
+}
+
+bool TypeTranslator::isSingleRowOrSingleColMatrix(QualType type,
```

Comment:
We have defined quite a few matrix probing methods. Opportunity to unify them a little bit?